src/components/SingleArticle.jsx

```diff
@@ -1,5 +1,30 @@
+import { useEffect, useState } from "react";
+import {useParams} from "react-router-dom";
+import {getArticleById} from "../utils/utils";
+
 const SingleArticle = () => {
-    return <h2>A single article</h2>
+    const {article_id} = useParams();
+    const [article, setArticle] = useState([]);
+    useEffect(()=>{
+        getArticleById(article_id)
+        .then((article) => {
+            setArticle(article)
+        })
+    }, [])
+    
+    return (
+        <div id="single-article-page">
+            <section id="single-article">
+                <p className="single-article-author">{article.author}</p>
+                <p className="single-article-date">{article.created_at}</p>
+                <h3 className="single-article-title">{article.title}</h3>
```

Comment:
Looks like you've jumped from h1 to h3 here which should be avoided. If the h2 is too big you'll be able to style it later with a class.